sceval_frontend/src/containers/SensorModule.tsx

```diff
@@ -224,17 +224,59 @@ export const SensorModule: React.FC<SensorModuleProps> = (props: SensorModulePro
                                     val: sensor.value
                                 });
                                 if (index === wsData.length - 1) { // if we have gone through all RT data:
-                                    const sortedRTData = rtData.sort((a: any, b: any) => {
-                                        if (a.type < b.type) {
-                                            return -1;
-                                        }
-                                        if (a.type > b.type) {
-                                            return 1;
-                                        }
-                                        return 0;
-                                    });
-                                    sensorContext.actions.setRTValues(rtNumbers);                        
-                                    setActiveSensors(sortedRTData); // set real time data
+                                    let updatedSensorData: any = activeSensors;
+                                    if (activeSensors) {
+                                    rtData.forEach((newSensor: any) => {
+                                        // filter and check if activeSensors exists
+                                            const dataExists = activeSensors.filter((onlineSensor: any): boolean => {
+                                                return onlineSensor.type === newSensor.type;
+                                            });
+                                            // if the sensor already has previous data, update it
+                                            let updatedActiveArray: any = activeSensors;
+                                            if (dataExists.length === 1) {
+                                                updatedActiveArray.forEach((updatedSensor: any) => {
+                                                    if (updatedSensor.type === newSensor.type) {
+                                                        updatedSensor.rtValue = newSensor.rtValue;
+                                                        setActiveSensors(updatedActiveArray.sort((a: any, b: any) => {
```

Comment:
@mattthewong, code blocks from line 240 - 249, 254 - 263, and 267 - 278 are duplicate. I think you can refactor this and move most of those code out of the if (activeSensors) {} else {} and put them after that if/else block.

Reply:
@mattthewong, updatedActiveArray.sort(...) will update the updatedActiveArray's item order so calling this inside of updatedActiveArray.forEach(...) might cause issue.

Reply:
Yep. Let me clean this up.